packages/skeleton-react/src/lib/components/Segment/Segment.tsx

```diff
@@ -1,113 +1,102 @@
 'use client';
 
-import React, { createContext, useContext, useEffect, useState } from 'react';
+import React, { createContext, useContext, useId } from 'react';
+import * as radio from '@zag-js/radio-group';
+import { useMachine, normalizeProps } from '@zag-js/react';
 import type { SegmentContextState, SegmentProps, SegmentItemsProps } from './types.js';
+import { noop } from '$lib/internal/noop.js';
 
 // Contexts ---
 
 export const SegmentContext = createContext<SegmentContextState>({
-	value: '',
-	name: '',
-	onSelectionHandler: () => {}
+	api: {} as ReturnType<typeof radio.connect>
 });
 
 // Components ---
 
 const SegmentRoot: React.FC<SegmentProps> = ({
-	value = '',
-	name = '',
 	// Root
 	base = 'inline-flex items-stretch overflow-hidden',
 	background = 'preset-outlined-surface-200-800',
 	border = 'p-2',
-	flexDirection = 'flex-row', // vertical: flex-col
 	gap = 'gap-2',
 	padding = '',
 	rounded = 'rounded-container',
 	width = '',
 	classes = '',
+	// Indicator
+	indicatorBase = 'top-[var(--top)] left-[var(--left)] w-[var(--width)] h-[var(--height)]',
+	indicatorBg = 'preset-filled',
+	indicatorRounded = 'rounded',
+	indicatorClasses = '',
 	// Events
-	onChange,
+	onValueChange = noop,
```

Comment:
Do we really want to explicitely define events like this when zag already does this for you in `...zagProps`? The only benefit we have is that it will be `onValueChange(value)` instead of `onValueChange({ value })`. Doesn't seem worth the trouble IMHO.

Reply:
I was using the Accordion as reference, which I think still utilizes a custom `onValueChange`. Maybe double check that?

Reply:
Per Discord - by moving this to the overflow `zagProps` end users run into two issues:

1. They can no longer use the `useState` setter directly (since this returns `{ value }`
2. They would be required to set the return value to `ValueChangeDetails` - which comes from Zag

I've kept the custom prop for now unless you have a better idea.

Reply:
Yeah this was my concern aswell, we can just use a custom callback indeed. I'll resolve this, was just wondering about this.